spaceship/lib/spaceship/client.rb

```diff
@@ -34,7 +34,31 @@ class InvalidUserCredentialsError < StandardError; end
     # Raised when no user credentials were passed at all
     class NoUserCredentialsError < StandardError; end
 
-    class UnexpectedResponse < StandardError; end
+    class UnexpectedResponse < StandardError
+      class << self
+        def handle_response!(response)
+          if response.body['userString']
+            require 'cgi'
+
+            result_string = CGI.unescapeHTML(response.body['resultString'])
+            user_string = CGI.unescapeHTML(response.body['userString'])
+            user_error!("Apple returned an unexpected response:\n\t#{result_string}\n\t#{user_string}")
+          else
+            raise UnexpectedResponse, response.body
+          end
+        end
+
+        def user_error!(msg)
+          require 'colored'
+
+          if $verbose # with stack trace
+            raise UnexpectedResponse, "[!] #{msg}".red
+          else # without stack trace
+            abort "\n[!] #{msg}".red
```

Comment:
This will require additional testing: Are those exceptions properly rescued by other classes around fastlane? The printing out of the stack trace is handled on a different layer (in the UI of fastlane), not in `spaceship`, which is a flexible library that developers can use to build their own tools.